x-pack/plugins/observability_ai_assistant/server/service/client/index.ts

```diff
@@ -0,0 +1,227 @@
+  }
+
+  private getConversationWithMetaFields = async (
+    conversationId: string
+  ): Promise<SearchHit<Conversation> | undefined> => {
+    const response = await this.dependencies.esClient.search<Conversation>({
+      index: this.dependencies.resources.aliases.conversations,
+      query: {
+        bool: {
+          filter: [...this.getAccessQuery(), { term: { 'conversation.id': conversationId } }],
+        },
+      },
+      size: 1,
+      terminate_after: 1,
+    });
+
+    return response.hits.hits[0];
+  };
+
+  private getConversationUpdateValues = (now: string) => {
+    return {
+      conversation: {
+        last_updated: now,
+      },
+      user: this.dependencies.user,
+      namespace: this.dependencies.namespace,
+    };
+  };
+
+  get = async (conversationId: string): Promise<Conversation | undefined> => {
+    return (await this.getConversationWithMetaFields(conversationId))?._source;
+  };
+
+  delete = async (conversationId: string): Promise<void> => {
+    const conversation = await this.getConversationWithMetaFields(conversationId);
+
+    if (!conversation) {
+      throw notFound();
+    }
+
+    await this.dependencies.esClient.delete({
+      id: conversation._id,
+      index: conversation._index,
+    });
+  };
+
+  chat = async ({
+    messages,
+    connectorId,
+  }: {
+    messages: Message[];
+    connectorId: string;
+  }): Promise<IncomingMessage> => {
+    const messagesForOpenAI: ChatCompletionRequestMessage[] = compact(
+      messages.map((message) => {
+        if (message.message.role === MessageRole.Event) {
+          return undefined;
+        }
+        const role =
+          message.message.role === MessageRole.Elastic ? MessageRole.User : message.message.role;
+
+        return {
+          role,
+          content: message.message.content,
+          function_call: isEmpty(message.message.function_call?.name)
+            ? undefined
+            : omit(message.message.function_call, 'trigger'),
+          name: message.message.name,
+        };
+      })
+    );
+
+    const connector = await this.dependencies.actionsClient.get({
+      id: connectorId,
+    });
+
+    const request: Omit<CreateChatCompletionRequest, 'model'> & { model?: string } = {
+      ...(connector.config?.apiProvider === OpenAiProviderType.OpenAi ? { model: 'gpt-4' } : {}),
+      messages: messagesForOpenAI,
+      stream: true,
+    };
+
+    const executeResult = await this.dependencies.actionsClient.execute({
+      actionId: connectorId,
+      params: {
+        subAction: 'stream',
+        subActionParams: {
+          body: JSON.stringify(request),
+          stream: true,
+        },
+      },
+    });
+
+    if (executeResult.status === 'error') {
+      throw internal(`${executeResult?.message} - ${executeResult?.serviceMessage}`);
+    }
+
+    return executeResult.data as IncomingMessage;
+  };
+
+  find = async (options?: { query?: string }): Promise<{ conversations: Conversation[] }> => {
```

Comment:
I think you are missing passing this option param to the actual elasticsearch query at the bottom?

Reply:
I'm not using it right now 😄 I'll figure that out later. It probably shouldn't be in the PR, but I wrote it already, and didn't want to spend much time removing it. Feel free to skip over it if it's taking too much time, although I appreciate any feedback :)